ixwebsocket/IXWebSocketTransport.cpp

```diff
@@ -176,33 +202,51 @@ namespace ix
         _onCloseCallback = onCloseCallback;
     }
 
-    // Only consider send time points for that computation.
-    // The receive time points is taken into account in Socket::poll (second parameter).
-    bool WebSocketTransport::heartBeatPeriodExceeded()
+    // Only consider send PING time points for that computation.
+    bool WebSocketTransport::pingIntervalExceeded()
     {
-        std::lock_guard<std::mutex> lock(_lastSendTimePointMutex);
+        if (_pingIntervalSecs <= 0)
+            return false;
+
+        std::lock_guard<std::mutex> lock(_lastSendPingTimePointMutex);
         auto now = std::chrono::steady_clock::now();
-        return now - _lastSendTimePoint > std::chrono::seconds(_heartBeatPeriod);
+        return now - _lastSendPingTimePoint > std::chrono::seconds(_pingIntervalSecs);
+    }
+
+    bool WebSocketTransport::pingTimeoutExceeded()
+    {
+        if (_pingTimeoutSecs <= 0)
+            return false;
+
+        std::lock_guard<std::mutex> lock(_lastReceivePongTimePointMutex);
+        auto now = std::chrono::steady_clock::now();
+        return now - _lastReceivePongTimePoint > std::chrono::seconds(_pingTimeoutSecs);
     }
 
     void WebSocketTransport::poll()
     {
+        if (_readyState == OPEN)
+        {
+            // if (1) ping tiemout is enabled and (2) duration exceeded the maximum delay, then close the connection
+            if (pingTimeoutExceeded())
+            {
+                close(1011, "Ping timeout");
+            }
+            // If (1) ping is enabled and (2) send for a duration exceeding our ping interval, send a ping to the server.
```

Comment:
That comment is not accurate anymore. What about making that:

```
// If (1) ping is enabled and no ping has been sent for a duration 
// exceeding our ping interval, send a ping to the server.
```

Reply:
Yes